### DISCUSSION TITLE
Help Desk software beyond GitHub Discussions and Issue Boards

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/riverscapes-website-jekyll#52] [Ideas]
@philipbaileynar, should we be considering some sort of helpdesk software beyond GitHub discussions and issue boards? Specifically, I am thinking from the new "[Support](https://docs.google.com/document/d/1g4-cvi-o34B3y0-mH5m91N5qTfquHlYqLfyVbWpQsvM/edit?usp=sharing)" Page for the RC site. I know we've made a big deal about a stupid email address, which I am still very opposed to. I think we can funnel and direct traffic much better with some more standard types of support. Maybe open-source versions exist... 

I am thinking that for the application support specialist team (e.g. @jtgilbert and @dominique-shore) will be our first line of contact and if we have something a little more professional as a first line of defense then what we get for free in clunky GitHub experience (great for developers, shit for endusers). I am comfortable paying for this service too. Features that get my attention:
- Deflect and direct common user questions
- Gather insights from common queries
- Automate routine workflows
- Offer Chat service that can easily be turned into tickets
- Offer real-time assistance remotely (instead of always having to make a Google Meet)
- Create tickets to track via email
- Create tickets via website forms
- Create tickets on the fly
- Assign and prioritize tickets and keep track for the application support team (separate from the developer teams)
 it might be nice to have a chat system, tied to tickets we track. I'm also thinking it might be a little bit of a cleaner line of defense between too much clutter and traffic on our GitHub sites for model, application specific traffic. 

A quick search and I found:

- [Odoo](https://www.odoo.com/app/helpdesk?utm_source=google&utm_medium=cpc&utm_campaign=NA-EN-Helpdesk&utm_term=open%20source%20helpdeskutm_gclid=CjwKCAjw6vyiBhB_EiwAQJRophIVOCHDcDNhsGcHN2RZijvSWWOQUwJKmv6tbvIM8GtctjvRqbXKKxoCDckQAvD_BwE&gclid=CjwKCAjw6vyiBhB_EiwAQJRophIVOCHDcDNhsGcHN2RZijvSWWOQUwJKmv6tbvIM8GtctjvRqbXKKxoCDckQAvD_BwE)
- [Freshdesk](https://www.freshworks.com/freshdesk/helpdesk-software-bex/)
- [ServiceDesk+](https://www.manageengine.com/products/service-desk/)

I'm not sure if any of these have API integration with GitHub or Zappier can be used? @MattReimer any suggestions or thoughts.

**Comment by MattReimer** [comment]
Kind of a big subject and customer-facing support isn't something I know super well. I haven't used any of these but if they're modern services they probably integrate with modern tools. 

I think the questions I'm curious to get answered are:

- What is the scope of this? What products specifically are we trying to support? All of riverscapes? Just the python? Website issues? Exchange issues?

- Whose job is it to maintain it (and it can't be Joe, Philip or Matt)? A ticketing system where tickets aren't resolved and build up over time is more of a liability than an asset. If we decide to use this it MUST be someone's job to check it and keep it clean. That person should probably have 30-60 minutes a day to spend on nothing but sorting out issues, asking questions to developers and communicating back to the people submitting them. 

Developers work through Github tickets and this system won't change that. Our developers aren't always great about tagging issues in their github issues or tagging them correctly or resolving them. Any integration between Github and one of these services is only as good as the diligence of the people who use it and I see a real potential for orphaned, disconnected, stagnant issues etc. 

Our GH ticket quality, tagging and PR hygiene varies pretty heavily with the production-quality of our repos and the experience level of the people who work on them and that's one of the important signals a system like this needs to be effective so that's probably a pretty big challenge to rolling this out.

Happy to discuss more.

**Reply by joewheaton** [reply]
> Kind of a big subject and customer-facing support isn't something I know super well. I haven't used any of these but if they're modern services they probably integrate with modern tools.
> 

I don't have all the answers, but here's a few:


> I think the questions I'm curious to get answered are:
> 
> * What is the scope of this? What products specifically are we trying to support? All of riverscapes? Just the python? Website issues? Exchange issues?

All of it. Main focus is supporting RC Applications, and RC Production Grade Models, but also lesser extent supporting developers who want to get their models riverscape compliant. This is mainly for consumers and basic users.  This is the job of the Application Support Specialists that @jtgilbert and I will supervise. We are going to have at least 1.5 to start and more if demand warrants it. 

> * Whose job is it to maintain it (and it can't be Joe, Philip or Matt)? A ticketing system where tickets aren't resolved and build up over time is more of a liability than an asset. If we decide to use this it MUST be someone's job to check it and keep it clean. That person should probably have 30-60 minutes a day to spend on nothing but sorting out issues, asking questions to developers and communicating back to the people submitting them.

Again, it is not the three of us. It is the job of the Application Support Specialists, supervised by @jtgilbert. The Application Support Specialist full time job is just this. Support. 

> 
> Developers work through Github tickets and this system won't change that. Our developers aren't always great about tagging issues in their github issues or tagging them correctly or resolving them. Any integration between Github and one of these services is only as good as the diligence of the people who use it and I see a real potential for orphaned, disconnected, stagnant issues etc.

Agreed. I see @jtgilbert being the main liaison there. Partly to shelter you @MattReimer. The Application Support Specialists will be provident enough to filter and flag whether something really is a bug and raise it in GitHub. After that, it is job of you all and @jtgilbert, but App Support Specialist will keep user with issue abreast of developments (even if, thank you, we reported, we have no budget or time to fix it right now, but it is on the list).

> 
> Our GH ticket quality, tagging and PR hygiene varies pretty heavily with the production-quality of our repos and the experience level of the people who work on them and that's one of the important signals a system like this needs to be effective so that's probably a pretty big challenge to rolling this out.
> 

I don't think it matters. Hygiene is a function of budget, time and quality of developers. Period. We know what our Applications are we are serious about and our Production Grade tools. The rest (e.g. Fluvial Corridor Toolbox) will vary with the quality of those developers and contributors. That is fine. That really only impacts a very, very, very miniscule small audience. My purpose in proposing help desk software is to have something for dealing with the masses of users we're trying to create and have a line of defense between you and them.

> Happy to discuss more.

**Comment by joewheaton** [comment]
@dominique-shore I heard from @philipbaileynar that Freshdesk is the solution we're going to try. Can you provide a few updates here? Then we'll mark that as our solution.

**Comment by dominique-shore** [comment]
@joewheaton I looked into a couple different HelpDesk software solutions that are good multi-channel options. FreshDesk & HelpScout seemed like the best options. [Here's a link ](https://docs.google.com/document/d/1itYSruCRuOGi_9Tz2DvSUNZa0SNiyYYK/edit?usp=sharing&ouid=116719902907923003412&rtpof=true&sd=true)to a comparison of the different options. 

The different between them most relevant is that FreshDesk supports integration with GitHub, and HelpScout does not. You can get a Zapier app to have integrations between HelpScout & FreshDesk. FreshDesk also supports a Forum on the knowledge base, and I don't believe HelpScout does.
